server/application-server/src/main/java/de/tum/in/www1/hephaestus/leaderboard/SlackMessageService.java

```diff
@@ -0,0 +1,133 @@
+    @Autowired
+    private LeaderboardService leaderboardService;
+
+    public void sendMessage(String channelId, List<LayoutBlock> blocks, String fallback)
+            throws IOException, SlackApiException {
+        ChatPostMessageRequest request = ChatPostMessageRequest.builder()
+                .channel(channelId)
+                .blocks(blocks)
+                .text(fallback) // used for example in notifications
+                .build();
+
+        ChatPostMessageResponse response = slackApp.client().chatPostMessage(request);
+
+        if (!response.isOk()) {
+            logger.error("Failed to send message to Slack channel: " + response.getError());
+        }
+    }
+
+    private List<LeaderboardEntryDTO> getTop3SlackReviewers() {
+        LocalDate after = LocalDate.now().minusDays(8);
+        LocalDate before = LocalDate.now().minusDays(1);
+        var leaderboard = leaderboardService.createLeaderboard(Optional.of(after), Optional.of(before),
+                Optional.empty());
+        return leaderboard.subList(0, Math.min(3, leaderboard.size()));
+    }
+
+    @Scheduled(cron = "${slack.cronSchedule}")
+    public void sendScheduledLeaderboard() {
+        if (!runScheduledMessage) {
+            return;
+        }
+
+        var top3reviewers = getTop3SlackReviewers();
+        logger.info("Top 3 Accounts of the last week: " + top3reviewers);
+
+        logger.info("Sending scheduled message to Slack channel...");
+        List<LayoutBlock> blocks = asBlocks(
+                header(header -> header.text(
+                        plainText(pt -> pt.text(":newspaper: Developer Reviews of the last week :newspaper:")))),
+                divider(),
+                context(context -> context
+                        .elements(List.of(markdownText("*October 31, 2024* | Hephaestus Announcement")))),
+                section(section -> section.text(markdownText(
+                        "Another *review leaderboard* has concluded. You can check out your placement at <https://hephaestus.ase.cit.tum.de/|hephaestus.ase.cit.tum.de>. The score is a weighted heuristic based on the complexities of the reviewed PRs (similar to the old algorithm)."))),
+                section(section -> section.text(markdownText(
+                        "If you would like to check out more leaderboards or statistics, head over to last week's leaderboard or this month's leaderboard."))),
+                section(section -> section.text(markdownText(
+                        "The top 3 reviewers of the last week are:"))),
+                section(section -> section.text(markdownText(
+                        IntStream.range(0, top3reviewers.size())
+                                .mapToObj(i -> ("• *" + (i + 1) + ". @" + top3reviewers.get(i).user().name() + " *"))
+                                .reduce((a, b) -> a + "\n" + b).orElse(""))))
+
+        );
+        try {
+            sendMessage(channelId, blocks, "Developer Reviews of the last week");
```

Comment:
I think we can also remove `Developer` here? How will this be displayed?

Reply:
This is basically a fallback text for very small UI elements. I believe this is used for certain kind of notifications, which cannot display the original message.
I just removed `Developer` from it. I will test it again as soon as we the bot gets the `users:read`-permission.